data-runtime/src/main/java/io/micronaut/data/runtime/intercept/AbstractQueryInterceptor.java

```diff
@@ -523,24 +523,20 @@ protected Object instantiateEntity(@NonNull Class<?> rootEntity, @NonNull Map<St
 
         BeanWrapper<Object> wrapper = BeanWrapper.getWrapper(instance);
         Collection<? extends PersistentProperty> persistentProperties = entity.getPersistentProperties();
-        for (PersistentProperty prop : persistentProperties) {
-            if (!prop.isReadOnly() && !prop.isGenerated()) {
-                String propName = prop.getName();
-                if (parameterValues.containsKey(propName)) {
-
-                    Object v = parameterValues.get(propName);
-                    if (v == null && !prop.isOptional()) {
-                        throw new IllegalArgumentException("Argument [" + propName + "] cannot be null");
-                    }
-                    wrapper.setProperty(propName, v);
-                } else if (prop.isRequired()) {
-                    final Optional<Object> p = wrapper.getProperty(propName, Object.class);
-                    if (!p.isPresent()) {
-                        throw new IllegalArgumentException("Argument [" + propName + "] cannot be null");
-                    }
+        PersistentProperty identity = entity.getIdentity();
```

Comment:
Try to match identity fields as well, persistentProperties that were used before didn't contain identity fields. It may be needed when not generated/readonly